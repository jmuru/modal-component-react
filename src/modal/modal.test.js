import React from 'react';
import TestRenderer from 'react-test-renderer';
import Modal from './modal'

describe('Modal', () => {
    test('should match snapshot when body is a string', () => {
        const modal = TestRenderer.create(<Modal title={'title'} subtitle={'subtitle'} modalBody={'owdfhlosdfhlsdhflsdhflsh'} />);
        const modalJSON = modal.toJSON();
        expect(modalJSON).toMatchSnapshot();
    });

    test('should match snapshot when body is a html element', () => {
        const modal = TestRenderer.create(<Modal title={'title'} subtitle={'subtitle'} modalBody={<div>hello world</div>} />);
        const modalJSON = modal.toJSON();
        expect(modalJSON).toMatchSnapshot();
    });
});
