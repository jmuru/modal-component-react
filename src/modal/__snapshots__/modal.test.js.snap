// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should match snapshot when body is a html element 1`] = `
<div
  class="modal-shadow"
  onClick={undefined}
>
  <div
    className="modal-container"
  >
    <div>
      <div
        className="modal-title-container"
      >
        <h1
          className="modal-title"
        >
          title
        </h1>
      </div>
      <div
        className="modal-subtitle-container"
      >
        <h4>
          subtitle
        </h4>
      </div>
      <div
        className="modal-body-container"
      >
        <div>
          hello world
        </div>
      </div>
    </div>
    <button
      onClick={undefined}
    >
      this is the close modal button
    </button>
  </div>
</div>
`;

exports[`Modal should match snapshot when body is a string 1`] = `
<div
  class="modal-shadow"
  onClick={undefined}
>
  <div
    className="modal-container"
  >
    <div>
      <div
        className="modal-title-container"
      >
        <h1
          className="modal-title"
        >
          title
        </h1>
      </div>
      <div
        className="modal-subtitle-container"
      >
        <h4>
          subtitle
        </h4>
      </div>
      <div
        className="modal-body-container"
      >
        <p>
          owdfhlosdfhlsdhflsdhflsh
        </p>
      </div>
    </div>
    <button
      onClick={undefined}
    >
      this is the close modal button
    </button>
  </div>
</div>
`;
